core/runtime/tests/jsrealm.rs

```diff
@@ -177,3 +179,39 @@ fn es_snapshot() {
     assert_eq!(test_export.to_rust_string_lossy(scope).as_str(), "bar");
   }
 }
+
+#[test]
+fn lazy() {
+  static CALLED: AtomicBool = AtomicBool::new(false);
+
+  deno_core::extension!(
+    lazy_ext,
+    options = {
+      a: String,
+      b: bool,
+    },
+    state = |_state, _options| {
+      CALLED.store(true, Ordering::Relaxed);
+    },
+  );
+
+  let extensions = vec![lazy_ext::lazy_init()];
+
+  let runtime = JsRuntime::new(RuntimeOptions {
+    extensions,
+    ..Default::default()
+  });
+
+  assert!(matches!(
+    runtime.lazy_init_extensions(vec![]),
```

Comment:
I wonder if we should have some sort of sanity check - if you had lazy init extensions and didn't call `runtime.lazy_init_extensions()` before trying to execute code or something then we error out

Reply:
Though this API is very exotic and for the most advanced users that should read docs carefully and uphold the invariants themselves 😆